next/components/mdx/mdx.tsx

```diff
@@ -0,0 +1,279 @@
+        return contents;
+    }
+
+    return <HashAnchor id={id}>{contents}</HashAnchor>;
+}
+
+export function H3({
+    children,
+    ...rest
+}: React.DetailedHTMLProps<
+    React.HTMLAttributes<HTMLHeadingElement>,
+    HTMLHeadingElement
+>): JSX.Element {
+    const id = generateSlug({ level: 'example', children });
+
+    const contents = (
+        <Title {...rest} id={id} size={5} headingLevel={3} className="mt5 mb2">
+            {children}
+        </Title>
+    );
+
+    if (!id) {
+        return contents;
+    }
+
+    return <HashAnchor id={id}>{contents}</HashAnchor>;
+}
+
+export function H4({
+    children,
+    ...rest
+}: React.DetailedHTMLProps<
+    React.HTMLAttributes<HTMLHeadingElement>,
+    HTMLHeadingElement
+>): JSX.Element {
+    const id = generateSlug({ level: 'example', children });
+
+    return (
+        <Title {...rest} id={id} size={6} headingLevel={4} className="mt5 mb1">
+            {children}
+        </Title>
+    );
+}
+
+export function P(p: Parameters<typeof Text>[0]): JSX.Element {
+    return <Text {...p} className={`mb3 black-300 ${styles.readingWidth}`} />;
+}
+
+export const InlineCode = ({
+    shouldCopyToClipboard = false,
+    children,
+    theme = 'default',
+}: {
+    shouldCopyToClipboard?: boolean;
+    children?: React.ReactNode;
+    theme?: 'plain' | 'default';
+}): JSX.Element => {
+    const plainStyles = {
+        fontFamily: tokens.tpFontFamilyMonospace,
+        fontSize: '95%',
+    };
+
+    const extendedStyles = {
+        background: '#f5f7f7',
+        padding: '1px 4px',
+        color: tokens.tpColorBlack,
+        borderRadius: '5px',
+    };
+
+    let inlineStyles = plainStyles;
+
+    if (theme !== 'plain') {
+        inlineStyles = { ...plainStyles, ...extendedStyles };
+    }
+
+    return shouldCopyToClipboard ? (
+        <CopyToClipboard text={children} className={styles.inlineCodeClipboard}>
+            <code style={inlineStyles}>{children}</code>
+        </CopyToClipboard>
+    ) : (
+        <code style={inlineStyles}>{children}</code>
+    );
+};
+
+export function Pre(
+    p: React.DetailedHTMLProps<React.HTMLAttributes<HTMLPreElement>, HTMLPreElement>,
+): JSX.Element {
+    return <pre {...p} />;
+}
+
+export function LI(p: Parameters<typeof Text>[0]): JSX.Element {
+    return (
+        <ListItem>
+            <Text elementName="div" className={`black-300 ${styles.readingWidth}`} {...p} />
+        </ListItem>
+    );
+}
+
+export function OL(p: Parameters<typeof List>[0]): JSX.Element {
+    return (
+        <div className="mb3 ml4">
+            <List theme="decimal" {...p} />
+        </div>
+    );
+}
+
+export function UL(p: Parameters<typeof List>[0]): JSX.Element {
+    return (
+        <div className="mb3 ml4">
+            <List {...p} />
+        </div>
+    );
+}
+
+export function Code(
+    p: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>,
+): JSX.Element {
+    // const language = p.className
```

Comment:
intentional?

Reply:
Yeah, this was causing some issues but I wanted to move on and fix in a future CR. I'll fix once I get to the point of getting the component documentation pages to work.